### PR TITLE
Automated cherry pick of #71597: Add additional err check for cloudProvider.InstanceNotFound

### PR DESCRIPTION
**Pull request by marc-sensenich**
Cherry pick of #71597 on release-1.12.

#71597: Add additional err check for cloudProvider.InstanceNotFound

### CODE REVIEW

File: pkg/cloudprovider/providers/azure/azure_instances.go [hunk 1]
```diff
@@ -23,10 +23,9 @@ import (
 	"strings"
 
 	"k8s.io/api/core/v1"
-	"k8s.io/kubernetes/pkg/cloudprovider"
-
-	"github.com/golang/glog"
 	"k8s.io/apimachinery/pkg/types"
+	cloudprovider "k8s.io/cloud-provider"
```
Comment (feiskyer) [inline pkg/cloudprovider/providers/azure/azure_instances.go]: could you change this back to `"k8s.io/kubernetes/pkg/cloudprovider"`

Comment (marc-sensenich) [reply pkg/cloudprovider/providers/azure/azure_instances.go]: Yeah, I'm going to close this PR out and attempt to do a cherry pick in a new PR. My IDE did some opinionated changes

Comment (marc-sensenich) [reply pkg/cloudprovider/providers/azure/azure_instances.go]: I fixed it on this pull request

**Pull request merged**